src/devices/SocketCan/SafeCanRawSocketHandle.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Net.Sockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+
+namespace Iot.Device.SocketCan
+{
+    internal class SafeCanRawSocketHandle : SafeHandle
+    {
+        public SafeCanRawSocketHandle(string networkInterface) : base(Interop.CreateCanRawSocket(networkInterface), true)
+        {
+        }
+
+        public override bool IsInvalid => (int)handle == -1;
+
+        protected override bool ReleaseHandle()
+        {
+            Interop.CloseSocket(handle);
```

Comment:
>Interop.CloseSocket(handle) [](start = 12, length = 27)

do you want to check IsValid before closing the socket?

Reply:
SafeHandle already checks that:
https://source.dot.net/#System.Private.CoreLib/shared/System/Runtime/InteropServices/SafeHandle.cs,220